portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -182,36 +175,39 @@ private String getUploadedFileUrl(HttpServletRequest request, String formInputEl
       if (nickname == null) {
         nickname = (String)entity.getProperty("userEmail");
       }
-      String imageUrl = (String)entity.getProperty("imageUrl");
-      Comment comment = new Comment((String)entity.getProperty("message"), nickname, imageUrl);
+      String imageString = (String)entity.getProperty("imageString");
+      Comment comment = new Comment((String)entity.getProperty("message"), nickname, imageString);
       
       comments.add(comment);
     }
 
     return comments;
   }
 
-  /** Creates Entity with a kind of Comment */
-  private Entity createCommentEntity(String newComment, String userEmail, String userId, String imageUrl) {
+  /** Creates Entity with a kind of Comment. */
+  private Entity createCommentEntity(String newComment, String userEmail, String userId, String imageString) {
     Entity commentEntity = new Entity("Comment");
     Timestamp timestamp = new Timestamp(System.currentTimeMillis());
 
     commentEntity.setProperty("message", newComment);
     commentEntity.setProperty("timestamp", timestamp.getTime());
     commentEntity.setProperty("userEmail", userEmail);
     commentEntity.setProperty("userNickname", getUserNickname(userId));
-    commentEntity.setProperty("imageUrl", imageUrl);
+    commentEntity.setProperty("imageString", imageString);
     return commentEntity;
   }
-  
+
+  /** Returns a String representing the text message entered by the user in commentForm.
+    * @throws UnsupportedEncodingException if the text contains invalid characters.
+    */
   private String getNewComment(HttpServletRequest request) throws UnsupportedEncodingException {
     String newComment = request.getParameter("new-comment");
    
     byte[] commentBytes = newComment.getBytes("UTF-8");
     return newComment;
   }
 
-  /** Returns the nickname of the user or null if no nickname was set by the user */
+  /** Returns the nickname of the user or null if no nickname was set by the user. */
```

Comment:
nit: Avoid punctuation and whitespace changes in unrelated PRs. You can always put 'white noise' changes in their own PR (ping on chat for instant approval, so you don't have to wait) and you're unlikely to have any serious conflicts anyway. A good rule of thumb is to never put refactoring changes (improve how the code is written/layed out with no behavior change) and changes (bug fixes, new feature implementation) in the same PR.

Otherwise, in this sea of small diffs it can be really hard to see what _actually_ changes.

Reply:
Thank you for the notice. The PR was meant only for refactoring changes and the bug was found while deploying the modified version. 
I understand that those changes only added noise to the bug fix and I will keep them separate in the future.